helpdesk/models.py

```diff
@@ -1012,6 +1004,54 @@ def get_markdown(self):
     def time_spent_formated(self):
         return format_time_spent(self.time_spent)
 
+    def time_spent_calculation(self):
+        "Returns timedelta according to rules settings."
+
+        # extract parameters from previous follow-up or ticket
+        try:
+            prev_fup_qs = self.ticket.followup_set.all()
+            if self.id:
+                prev_fup_qs = prev_fup_qs.filter(id__lt=self.id)
+            prev_fup = prev_fup_qs.latest("date")
+            earliest = prev_fup.date
+            prev_status = prev_fup.new_status
```

Comment:
If you are talking about *.filter(id__lt=self.id)*, yes, it will be *None* most of the time.

But let's say, you changed your OPENING_HOURS configuration and need all your previous FollowUps to be recalculated.
You may empty all your time_spent values, resave( and they will be correctly recalculated with self.id not being None anymore.